packages/kotti-ui/source/kotti-comment/KtComment.vue

```diff
@@ -95,75 +104,74 @@ export default {
 		KtCommentInput,
 	},
 	props: makeProps(KottiComment.propsSchema),
-	data() {
-		return {
-			showInlineReply: false,
-			replyToText: null,
-			replyToUserId: Number,
-			isInlineEdit: false,
-			inlineMessageValue: '',
+	setup(props, { emit }) {
+		const isInlineEdit = ref(false)
+		const showInlineReply = ref(false)
+		const inlineMessageValue = ref<string | null>(null)
+		const userBeingRepliedTo = ref<UserData | null>(null)
+
+		const handleDelete = (commentId: number | string, isInline?: boolean) => {
+			emit('delete', {
+				id: commentId,
+				parentId: isInline ? props.id : null,
+			})
 		}
-	},
-	computed: {
-		styleObject() {
-			return {
-				avatar: true,
-				'avatar-selected': this.selected,
-			}
-		},
-		inlineMessage() {
-			return this.inlineMessageValue || this.message
-		},
-		actionOptions() {
-			const options = []
-			if (this.isEditable)
-				options.push({
-					label: 'Edit',
-					onClick: () => {
-						this.inlineMessageValue = this.inlineMessage
-						this.isInlineEdit = true
-					},
-					type: 'EDIT',
-				})
-			if (this.isDeletable)
-				options.push({
-					label: 'Delete',
-					onClick: () => this.handleDelete(this.id),
-					type: 'DELETE',
+		return {
+			actionOptions: computed<Array<KottiButton.Props>>(() =>
+				compact([
+					props.isEditable
+						? {
+								label: 'Edit',
+								onClick: () => {
+									if (inlineMessageValue.value === null)
+										inlineMessageValue.value = props.message
+									isInlineEdit.value = true
+								},
+								type: KottiButton.Type.PRIMARY,
+						  }
+						: null,
+					props.isDeletable
+						? {
+								label: 'Delete',
+								onClick: () => handleDelete(props.id),
+								type: KottiButton.Type.DANGER,
+						  }
+						: null,
+				]),
+			),
+			cancelInlineEdit: () => {
+				inlineMessageValue.value = null
```

Comment:
you reset `inlineMessageValue` everytime you cancel the edit action

so just checking why you introduced this check: 
https://github.com/3YOURMIND/kotti/pull/562/commits/9bdc06b7e030544f1149bbd9957f44470b48c8a1#diff-9588d13c829fac8eaaab185abb19d1e652ba33ceec38a11da7f7fe5ac39d7924R126


it seems that every time you click `edit` you want to initialize the inlineEditMessage as `props.message`

but I guess you wanted to handle the case in the UI where the user clicks the `EDIT` while the edit is already active

so that you don't reset what they temporarily typed


I suggest the following then because it makes it clearer

convert the condition to `if(!isInlineEdit.value) return`
because it makes it clear that this is a redudant extra click

and then it won't execute the reset of the message to `props.message` OR the re-initialization of `isInlinedEdit` to `true`

Reply:
I thought it made more sense to force those options to be empty when `isInlineEdit.value` is `true`

Reply:
but you didn't force the edit option to not be there, with the condition: `if (inlineMessageValue.value === null)`

you were just guarding before resetting the edit value

which meant you actually prevented the bug that if the user clicks once then starts typing THEN clicks again on the option, it wouldn't have reset the already-typed message to `null`

so I basically said, I actually agree with what you did, I just think it might be more readable to have `if(isInlineEdit) return`

because if the flag that already shows the user is editing is true, then there is no need to reset